core/src/main/java/org/geysermc/geyser/registry/populator/BlockRegistryPopulator.java

```diff
@@ -263,6 +350,34 @@ private static void registerBedrockBlocks() {
             }
             builder.bedrockMovingBlock(movingBlockDefinition);
 
+            Map<JavaBlockState, CustomBlockState> nonVanillaStateOverrides = BlockRegistries.NON_VANILLA_BLOCK_STATE_OVERRIDES.get();
+            if (nonVanillaStateOverrides.size() > 0) {
+                // First ensure all non vanilla runtime IDs at minimum are air in case they aren't consecutive
+                Arrays.fill(javaToVanillaBedrockBlocks, minCustomRuntimeID, javaToVanillaBedrockBlocks.length, airDefinition);
+                Arrays.fill(javaToBedrockBlocks, minCustomRuntimeID, javaToBedrockBlocks.length, airDefinition);
+
+                for (Map.Entry<JavaBlockState, CustomBlockState> entry : nonVanillaStateOverrides.entrySet()) {
+                    GeyserBedrockBlock bedrockDefinition = customBlockStateDefinitions.get(entry.getValue());
+                    if (bedrockDefinition == null) {
+                        GeyserImpl.getInstance().getLogger().warning("Unable to find custom block for " + entry.getValue());
+                        continue;
+                    }
+
+                    JavaBlockState javaState = entry.getKey();
+                    int stateRuntimeId = javaState.javaId();
+
+                    String javaId = javaState.identifier();
+                    boolean waterlogged = javaId.contains("waterlogged=true");
```

Comment:
Should this be `javaState.waterlogged()`?

Reply:
https://github.com/GeyserMC/Geyser/pull/3505/commits/ba7708f25b97b5bc07c31a3be48b3f31cc1df6a0#diff-c0a896602d5d2af0f1332c0ccc71deea24882f7b0c8087f6df9928a7eb13382eR369